### PR TITLE
Update NGINXaaS API: 2024-09-01-preview

### PR DESCRIPTION
**Pull request by puneetsarna**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## PR Checklist

- [ ] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [ ] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [ ] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [ ] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [ ] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [ ] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [ ] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [ ] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_resource` - support for the `thing1` property [GH-00000]

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Comment by puneetsarna** [comment]
Depends on https://github.com/hashicorp/terraform-provider-azurerm/pull/27775

**Comment by puneetsarna** [comment]
Need to fix up property deprecation.

**Comment by puneetsarna** [comment]
This is a rough/incomplete draft for now. CC @stephybun

**Review by stephybun — CHANGES_REQUESTED** [review]
we Hey @puneetsarna

Before I dive into a detailed review here, I noticed that the acceptance tests for the configuration resource begun failing from the 15th of October. There is a related [PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/27734) open that attempts to fix this but it seems there are some issues around testing here.

This test passing is a pre-requisite to reviewing and merging this change since we rely on those to validate that the deprecation and changes in this PR have been done correctly and won't cause any additional issues or unexpected side effects for users.

Would you be able to work with the author of the PR linked above to get that test fixed?

**Comment by puneetsarna** [comment]
> Hey @puneetsarna
> 
> Before I dive into a detailed review here, I noticed that the acceptance tests for the configuration resource begun failing from the 15th of October. There is a related [PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/27734) open that attempts to fix this but it seems there are some issues around testing here.
> 
> This test passing is a pre-requisite to reviewing and merging this change since we rely on those to validate that the deprecation and changes in this PR have been done correctly and won't cause any additional issues or unexpected side effects for users.
> 
> Would you be able to work with the author of the PR linked above to get that test fixed?

Thanks @stephybun I took a look at that PR and have left some comments. Can you help review it as well please? then we can start on this PR also?

**Comment by stephybun** [comment]
@puneetsarna could you rebase this on top of main to pull in those nginx configuration test fixes?

**Comment by puneetsarna** [comment]
> @puneetsarna could you rebase this on top of main to pull in those nginx configuration test fixes?

@stephybun Done 👍

File: internal/services/nginx/nginx_configuration_resource.go [hunk 1]
```diff
@@ -270,18 +269,10 @@ func (m ConfigurationResource) Read() sdk.ResourceFunc {
...
-							configs = append(configs, ProtectedFile{
-								Content:     pointer.ToString(file.Content),
-								VirtualPath: pointer.ToString(file.VirtualPath),
-							})
-						}
-					}
-
-					if len(configs) > 0 {
-						output.ProtectedFile = configs
+						output.ProtectedFile = append(output.ProtectedFile, ProtectedFile{
```
Comment (puneetsarna) [inline internal/services/nginx/nginx_configuration_resource.go:273]: This logic is going to fail the provider test.

File: internal/services/nginx/nginx_configuration_data_source.go [hunk 2]
```diff
@@ -87,6 +82,29 @@ func (m ConfigurationDataSource) Attributes() map[string]*pluginsdk.Schema {
...
+						Sensitive:  true,
+					},
+
+					"virtual_path": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		}
```
Comment (stephybun) [inline internal/services/nginx/nginx_configuration_data_source.go:105]: It's helpful to target the `content` field specifically, makes it easier for the reviewer to see what will be removed in 5.0
```suggestion
		dataSource["protected_file"].Elem.(*pluginsdk.Resource).Schema["content"] = &pluginsdk.Schema{
			Type:     pluginsdk.TypeString,
			Computed: true,
			Sensitive: true,
			Deprecated: "the `content` property is deprecated and will be removed in v5.0 of the AzureRM Provider.",
		}
```

File: internal/services/nginx/nginx_configuration_resource.go [hunk 3]
```diff
@@ -36,10 +36,10 @@ type ProtectedFile struct {
...
 }
 
-func (c ProtectedFile) toSDKModel() nginxconfiguration.NginxConfigurationFile {
-	return nginxconfiguration.NginxConfigurationFile{
-		Content:     pointer.To(c.Content),
-		VirtualPath: pointer.To(c.VirtualPath),
+func (c ProtectedFile) toSDKModel() nginxconfiguration.NginxConfigurationProtectedFileRequest {
+	return nginxconfiguration.NginxConfigurationProtectedFileRequest{
+		Content:     pointer.FromString(c.Content),
+		VirtualPath: pointer.FromString(c.VirtualPath),
```
Comment (stephybun) [inline internal/services/nginx/nginx_configuration_resource.go:42]: This should remain as 
```suggestion
		Content:     pointer.To(c.Content),
		VirtualPath: pointer.To(c.VirtualPath),
```

Also shouldn't `content` be deprecated in the resource as well?

File: internal/services/nginx/nginx_configuration_resource.go [hunk 4]
```diff
@@ -59,24 +59,23 @@ func (c ConfigurationModel) toSDKFiles() *[]nginxconfiguration.NginxConfiguratio
...
-func (c ConfigurationModel) ToSDKModel() nginxconfiguration.NginxConfiguration {
-	req := nginxconfiguration.NginxConfiguration{
-		Name: pointer.To(defaultConfigurationName),
-		Properties: &nginxconfiguration.NginxConfigurationProperties{
-			RootFile: pointer.To(c.RootFile),
+func (c ConfigurationModel) ToSDKModel() nginxconfiguration.NginxConfigurationRequest {
+	req := nginxconfiguration.NginxConfigurationRequest{
+		Name: pointer.FromString(defaultConfigurationName),
+		Properties: &nginxconfiguration.NginxConfigurationRequestProperties{
+			RootFile: pointer.FromString(c.RootFile),
```
Comment (stephybun) [inline internal/services/nginx/nginx_configuration_resource.go:78]: Same here
```suggestion
		Name: pointer.To(defaultConfigurationName),
		Properties: &nginxconfiguration.NginxConfigurationRequestProperties{
			RootFile: pointer.To(c.RootFile),
```

File: internal/services/nginx/nginx_configuration_resource.go [hunk 5]
```diff
@@ -315,7 +306,10 @@ func (m ConfigurationResource) Update() sdk.ResourceFunc {
 				return fmt.Errorf("retrieving as nil for %v", *id)
 			}
 
-			upd := existing.Model
+			upd := nginxconfiguration.NginxConfigurationRequest{
+				Name: pointer.FromString(defaultConfigurationName),
+			}
+
```
Comment (stephybun) [inline internal/services/nginx/nginx_configuration_resource.go:312]: This might end up breaking the `Update` for the resource, what is the reason for this change?

File: internal/services/nginx/nginx_deployment_resource.go [hunk 6]
```diff
@@ -260,6 +253,17 @@ func (m DeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 		"tags": commonschema.Tags(),
 	}
 
+	if !features.FivePointOhBeta() {
+		resource["managed_resource_group"] = &pluginsdk.Schema{
+			Deprecated:   "The `managed_resource_group` field has been deprecated and will be removed in v5.0 of the AzureRM Provider.",
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ForceNew:     true,
```
Comment (stephybun) [inline internal/services/nginx/nginx_deployment_resource.go:262]: We should remove `ForceNew` on this so that users can remove this from their configs without triggering a recreation of the resource 
```suggestion
```

File: internal/services/nginx/nginx_deployment_resource.go [hunk 7]
```diff
@@ -260,6 +253,17 @@ func (m DeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 		"tags": commonschema.Tags(),
 	}
 
+	if !features.FivePointOhBeta() {
+		resource["managed_resource_group"] = &pluginsdk.Schema{
+			Deprecated:   "The `managed_resource_group` field has been deprecated and will be removed in v5.0 of the AzureRM Provider.",
```
Comment (stephybun) [inline internal/services/nginx/nginx_deployment_resource.go:258]: I think it would be good to specify that this property isn't supported in the API anymore, so users are aware that they can't expect the same behaviour as before despite the deprecation
```suggestion
			Deprecated:   "The `managed_resource_group` field isn't supported by the API anymore and has been deprecated and will be removed in v5.0 of the AzureRM Provider.",
```

**Review by stephybun — CHANGES_REQUESTED** [review]
I've taken another look over this @puneetsarna. Aside from the comments and questions left in-line the following points still need to be addressed:

- The tests probably need an update to remove these deprecated properties from the test configuration
- The resource and data source documentation needs to be updated
- The [5.0 upgrade guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/website/docs/5.0-upgrade-guide.html.markdown) needs to be updated with the deprecations

File: internal/services/nginx/nginx_configuration_data_source.go (hunk 2 above)
Comment (puneetsarna) [reply internal/services/nginx/nginx_configuration_data_source.go:105]: thanks for sharing this 👍

File: internal/services/nginx/nginx_configuration_resource.go (hunk 3 above)
Comment (puneetsarna) [reply internal/services/nginx/nginx_configuration_resource.go:42]: We still want the user to be able to provide the content of the file. It's the output that they get back will no longer be the case.

File: internal/services/nginx/nginx_configuration_resource.go (hunk 4 above)
Comment (puneetsarna) [reply internal/services/nginx/nginx_configuration_resource.go:78]: 👍

File: internal/services/nginx/nginx_deployment_resource.go (hunk 7 above)
Comment (puneetsarna) [reply internal/services/nginx/nginx_deployment_resource.go:258]: 👍

File: internal/services/nginx/nginx_configuration_resource.go (hunk 5 above)
Comment (stephybun) [reply internal/services/nginx/nginx_configuration_resource.go:322]: This is still an issue @puneetsarna

Comment (puneetsarna) [reply internal/services/nginx/nginx_configuration_resource.go:322]: @stephybun my apologies. my comments were in a pending state and I never submitted them. 🤦 

Can you please take a look at my response above.

**Review by stephybun — CHANGES_REQUESTED** [review]
@puneetsarna once my last comment has been resolved and the test failure fixed this should be good to go
```
Test ended in panic.
------- Stdout: -------
=== RUN   TestAccConfiguration_update
=== PAUSE TestAccConfiguration_update
=== CONT  TestAccConfiguration_update
------- Stderr: -------
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4deb000]
goroutine 10826 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/nginx.(*ConfigurationResource).Update.ConfigurationResource.Update.func1({0x8c6c078, 0xc002470cb0}, {0xc004417208, {0x8c6c200, 0xc004bc1800}, 0xc00243d000, 0x0, {0x8c6c238, 0xe52e5e0}})
  /opt/teamcity-agent/work/3337027aeff310bf/internal/services/nginx/nginx_configuration_resource.go:329 +0x2e0
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.func7({0x8c6c078, 0xc002470cb0}, 0xc002470cb0?, {0x733a760?, 0xc004417208?})
```

**Comment by puneetsarna** [comment]
@stephybun ran the tests from the latest push:

```shell
TF_ACC=1 go test -v ./internal/services/nginx -run=TestAccConfiguration_update -timeout 180m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccConfiguration_update
=== PAUSE TestAccConfiguration_update
=== CONT  TestAccConfiguration_update
--- PASS: TestAccConfiguration_update (486.47s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/nginx	489.158s
terraform-provider-azurerm $
terraform-provider-azurerm $
terraform-provider-azurerm $
(reverse-i-search)`':
terraform-provider-azurerm $ git push psarna +$(git br)
Everything up-to-date
terraform-provider-azurerm $ make acctests SERVICE='nginx' TESTARGS='-run=TestAccConfiguration_basic'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/nginx -run=TestAccConfiguration_basic -timeout 180m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccConfiguration_basic
=== PAUSE TestAccConfiguration_basic
=== CONT  TestAccConfiguration_basic
--- PASS: TestAccConfiguration_basic (484.23s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/nginx	486.865s
```

File: internal/services/nginx/nginx_configuration_resource.go (hunk 5 above)
Comment (puneetsarna) [reply internal/services/nginx/nginx_configuration_resource.go]: good point. As I did this, I realized that the rootFile is a required property in the request. I decided to keep those lines of code to retrieve the current rootFile from the service API, set it, and then move on to the logic to see if it has changed.

File: internal/services/nginx/nginx_configuration_resource.go [hunk 8]
```diff
@@ -272,14 +271,16 @@ func (m ConfigurationResource) Read() sdk.ResourceFunc {
...
-								Content:     pointer.ToString(file.Content),
-								VirtualPath: pointer.ToString(file.VirtualPath),
-							})
+						for _, protectedFile := range output.ProtectedFile {
+							if protectedFile.VirtualPath == pointer.ToString(file.VirtualPath) {
+								configs = append(configs, ProtectedFile{
+									Content:     protectedFile.Content,
+									VirtualPath: pointer.ToString(file.VirtualPath),
+								})
+								break
```
Comment (stephybun) [inline internal/services/nginx/nginx_configuration_resource.go:280]: The schema allows users to specify multiple protected files, I don't know whether this is a valid configuration from the perspective of the API but if it is and we're breaking out of the for loop here, then this will break for users that have more than one protected file specified in their config

**Review by stephybun — CHANGES_REQUESTED** [review]
@puneetsarna the tests look good, because we're making a breaking change behind the `features.FivePointOhBeta` flag the tests also need in pass in 5.0 mode which they're not. See failure below
```
Test ended in panic.
------- Stdout: -------
=== RUN   TestAccConfiguration_withCertificate
=== PAUSE TestAccConfiguration_withCertificate
=== CONT  TestAccConfiguration_withCertificate
------- Stderr: -------
panic: Invalid address to set: []string{"managed_resource_group"}
goroutine 2356 [running]:
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc003b34c00, {0x6f7e3d1, 0x16}, {0x7101740, 0xe530ba0})
  /opt/teamcity-agent/work/3337027aeff310bf/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource_data.go:233 +0x2b4
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.ResourceMetaData.Encode({0xc0008d2488, {0x8c6b880, 0xc003b73de8}, 0xc003b34c00, 0x0, {0x8c6b8b8, 0xe52e5c0}}, {0x6e02dc0, 0xc003b554a0})
```

**Comment by puneetsarna** [comment]
> @puneetsarna the tests look good, because we're making a breaking change behind the `features.FivePointOhBeta` flag the tests also need in pass in 5.0 mode which they're not. See failure below
> 
> ```
> Test ended in panic.
> ------- Stdout: -------
> === RUN   TestAccConfiguration_withCertificate
> === PAUSE TestAccConfiguration_withCertificate
> === CONT  TestAccConfiguration_withCertificate
> ------- Stderr: -------
> panic: Invalid address to set: []string{"managed_resource_group"}
> goroutine 2356 [running]:
> github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc003b34c00, {0x6f7e3d1, 0x16}, {0x7101740, 0xe530ba0})
>   /opt/teamcity-agent/work/3337027aeff310bf/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource_data.go:233 +0x2b4
> github.com/hashicorp/terraform-provider-azurerm/internal/sdk.ResourceMetaData.Encode({0xc0008d2488, {0x8c6b880, 0xc003b73de8}, 0xc003b34c00, 0x0, {0x8c6b8b8, 0xe52e5c0}}, {0x6e02dc0, 0xc003b554a0})
> ```

I was missing the tag `removedInNextMajorVersion` on the managed_resource_group field. 👍 

I ran the tests locally at 5.0 and they are passing now 👍

File: internal/services/nginx/nginx_configuration_resource.go (hunk 8 above)
Comment (puneetsarna) [reply internal/services/nginx/nginx_configuration_resource.go:280]: Based on an offline discussion:

There was a slight misunderstanding here which led to the comment. Another review of it revealed an issue with drift detection and @stephybun and I talked over a possible fix for it which has been pushed.

**Review by stephybun — COMMENTED** [review]
Thanks @puneetsarna! There are still a few test failures in 5.0 mode, once those are fixed this should be good to go!
```
Test ended in panic.
------- Stdout: -------
=== RUN   TestAccNginxConfigurationDataSource_basic
=== PAUSE TestAccNginxConfigurationDataSource_basic
=== CONT  TestAccNginxConfigurationDataSource_basic
------- Stderr: -------
panic: Invalid address to set: []string{"protected_file", "0", "content"}
goroutine 2231 [running]:
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc0029fbc80, {0x6ca2375, 0xe}, {0x6f80740, 0xc004f68b70})
  /opt/teamcity-agent/work/3337027aeff310bf/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource_data.go:233 +0x2b4
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.ResourceMetaData.Encode({0xc004e52488, {0x8c6ca20, 0xc001ed4678}, 0xc0029fbc80, 0x0, {0x8c6ca58, 0xe52f5c0}}, {0x6e03c40, 0xc00421d6e0})
  /opt/teamcity-agent/work/3337027aeff310bf/internal/sdk/resource_encode.go:31 +0x247
github.com/hashicorp/terraform-provider-azurerm/internal/services/nginx.(*ConfigurationDataSource).Read.ConfigurationDataSource.Read.func1({0x8c6c898, 0xc005137180}, {0xc004e52488, {0x8c6ca20, 0xc001ed4678}, 0xc0029fbc80, 0x0, {0x8c6ca58, 0xe52f5c0}})
  /opt/teamcity-agent/work/3337027aeff310bf/internal/services/nginx/nginx_configuration_data_source.go:163 +0x5b3
```
```
Test ended in panic.
------- Stdout: -------
=== RUN   TestAccNginxDeploymentDataSource_basic
=== PAUSE TestAccNginxDeploymentDataSource_basic
=== CONT  TestAccNginxDeploymentDataSource_basic
------- Stderr: -------
panic: Invalid address to set: []string{"managed_resource_group"}
goroutine 2163 [running]:
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc003ed7100, {0x6f7f3d1, 0x16}, {0x7102740, 0xe531ba0})
  /opt/teamcity-agent/work/3337027aeff310bf/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource_data.go:233 +0x2b4
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.ResourceMetaData.Encode({0xc00252ed88, {0x8c6ca20, 0xc005a2f2f0}, 0xc003ed7100, 0x0, {0x8c6ca58, 0xe52f5c0}}, {0x6e03d00, 0xc002a89040})
  /opt/teamcity-agent/work/3337027aeff310bf/internal/sdk/resource_encode.go:31 +0x247
github.com/hashicorp/terraform-provider-azurerm/internal/services/nginx.(*DeploymentDataSource).Read.DeploymentDataSource.Read.func1({0x8c6c898, 0xc003e9f960}, {0xc00252ed88, {0x8c6ca20, 0xc005a2f2f0}, 0xc003ed7100, 0x0, {0x8c6ca58, 0xe52f5c0}})
  /opt/teamcity-agent/work/3337027aeff310bf/internal/services/nginx/nginx_deployment_data_source.go:324 +0x898
```

```
Test ended in panic.
------- Stdout: -------
=== RUN   TestAccNginxDeploymentDataSource_autoscaling
=== PAUSE TestAccNginxDeploymentDataSource_autoscaling
=== CONT  TestAccNginxDeploymentDataSource_autoscaling
------- Stderr: -------
panic: Invalid address to set: []string{"managed_resource_group"}
goroutine 2122 [running]:
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc002bbac80, {0x6f7f3d1, 0x16}, {0x7102740, 0xe531ba0})
  /opt/teamcity-agent/work/3337027aeff310bf/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource_data.go:233 +0x2b4
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.ResourceMetaData.Encode({0xc004914908, {0x8c6ca20, 0xc0053f3680}, 0xc002bbac80, 0x0, {0x8c6ca58, 0xe52f5c0}}, {0x6e03d00, 0xc002a6d900})
  /opt/teamcity-agent/work/3337027aeff310bf/internal/sdk/resource_encode.go:31 +0x247
github.com/hashicorp/terraform-provider-azurerm/internal/services/nginx.(*DeploymentDataSource).Read.DeploymentDataSource.Read.func1({0x8c6c898, 0xc002835110}, {0xc004914908, {0x8c6ca20, 0xc0053f3680}, 0xc002bbac80, 0x0, {0x8c6ca58, 0xe52f5c0}})
  /opt/teamcity-agent/work/3337027aeff310bf/internal/services/nginx/nginx_deployment_data_source.go:324 +0x898
```

**Comment by puneetsarna** [comment]
> Thanks @puneetsarna! There are still a few test failures in 5.0 mode, once those are fixed this should be good to go!
> 
> ```
> Test ended in panic.
> ------- Stdout: -------
> === RUN   TestAccNginxConfigurationDataSource_basic
> === PAUSE TestAccNginxConfigurationDataSource_basic
> === CONT  TestAccNginxConfigurationDataSource_basic
> ------- Stderr: -------
> panic: Invalid address to set: []string{"protected_file", "0", "content"}
> goroutine 2231 [running]:
> github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc0029fbc80, {0x6ca2375, 0xe}, {0x6f80740, 0xc004f68b70})
>   /opt/teamcity-agent/work/3337027aeff310bf/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource_data.go:233 +0x2b4
> github.com/hashicorp/terraform-provider-azurerm/internal/sdk.ResourceMetaData.Encode({0xc004e52488, {0x8c6ca20, 0xc001ed4678}, 0xc0029fbc80, 0x0, {0x8c6ca58, 0xe52f5c0}}, {0x6e03c40, 0xc00421d6e0})
>   /opt/teamcity-agent/work/3337027aeff310bf/internal/sdk/resource_encode.go:31 +0x247
> github.com/hashicorp/terraform-provider-azurerm/internal/services/nginx.(*ConfigurationDataSource).Read.ConfigurationDataSource.Read.func1({0x8c6c898, 0xc005137180}, {0xc004e52488, {0x8c6ca20, 0xc001ed4678}, 0xc0029fbc80, 0x0, {0x8c6ca58, 0xe52f5c0}})
>   /opt/teamcity-agent/work/3337027aeff310bf/internal/services/nginx/nginx_configuration_data_source.go:163 +0x5b3
> ```
> 
> ```
> Test ended in panic.
> ------- Stdout: -------
> === RUN   TestAccNginxDeploymentDataSource_basic
> === PAUSE TestAccNginxDeploymentDataSource_basic
> === CONT  TestAccNginxDeploymentDataSource_basic
> ------- Stderr: -------
> panic: Invalid address to set: []string{"managed_resource_group"}
> goroutine 2163 [running]:
> github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc003ed7100, {0x6f7f3d1, 0x16}, {0x7102740, 0xe531ba0})
>   /opt/teamcity-agent/work/3337027aeff310bf/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource_data.go:233 +0x2b4
> github.com/hashicorp/terraform-provider-azurerm/internal/sdk.ResourceMetaData.Encode({0xc00252ed88, {0x8c6ca20, 0xc005a2f2f0}, 0xc003ed7100, 0x0, {0x8c6ca58, 0xe52f5c0}}, {0x6e03d00, 0xc002a89040})
>   /opt/teamcity-agent/work/3337027aeff310bf/internal/sdk/resource_encode.go:31 +0x247
> github.com/hashicorp/terraform-provider-azurerm/internal/services/nginx.(*DeploymentDataSource).Read.DeploymentDataSource.Read.func1({0x8c6c898, 0xc003e9f960}, {0xc00252ed88, {0x8c6ca20, 0xc005a2f2f0}, 0xc003ed7100, 0x0, {0x8c6ca58, 0xe52f5c0}})
>   /opt/teamcity-agent/work/3337027aeff310bf/internal/services/nginx/nginx_deployment_data_source.go:324 +0x898
> ```
> 
> ```
> Test ended in panic.
> ------- Stdout: -------
> === RUN   TestAccNginxDeploymentDataSource_autoscaling
> === PAUSE TestAccNginxDeploymentDataSource_autoscaling
> === CONT  TestAccNginxDeploymentDataSource_autoscaling
> ------- Stderr: -------
> panic: Invalid address to set: []string{"managed_resource_group"}
> goroutine 2122 [running]:
> github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc002bbac80, {0x6f7f3d1, 0x16}, {0x7102740, 0xe531ba0})
>   /opt/teamcity-agent/work/3337027aeff310bf/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource_data.go:233 +0x2b4
> github.com/hashicorp/terraform-provider-azurerm/internal/sdk.ResourceMetaData.Encode({0xc004914908, {0x8c6ca20, 0xc0053f3680}, 0xc002bbac80, 0x0, {0x8c6ca58, 0xe52f5c0}}, {0x6e03d00, 0xc002a6d900})
>   /opt/teamcity-agent/work/3337027aeff310bf/internal/sdk/resource_encode.go:31 +0x247
> github.com/hashicorp/terraform-provider-azurerm/internal/services/nginx.(*DeploymentDataSource).Read.DeploymentDataSource.Read.func1({0x8c6c898, 0xc002835110}, {0xc004914908, {0x8c6ca20, 0xc0053f3680}, 0xc002bbac80, 0x0, {0x8c6ca58, 0xe52f5c0}})
>   /opt/teamcity-agent/work/3337027aeff310bf/internal/services/nginx/nginx_deployment_data_source.go:324 +0x898
> ```

Thanks @stephybun. I had missed out on more `removedInNextMajorVersion` stuff. pushed up a fix for it and tests are passing with provider 5.0:

```shell
TF_ACC=1 go test -v ./internal/services/nginx -run=TestAccNginxConfigurationDataSource_basic -timeout 180m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccNginxConfigurationDataSource_basic
=== PAUSE TestAccNginxConfigurationDataSource_basic
=== CONT  TestAccNginxConfigurationDataSource_basic


--- PASS: TestAccNginxConfigurationDataSource_basic (439.64s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/nginx	442.521s

-------------------------------------------------------------------------------------------------
=== CONT  TestAccNginxDeploymentDataSource_basic
--- PASS: TestAccNginxDeploymentDataSource_basic (405.34s)
--- PASS: TestAccNginxDeploymentDataSource_autoscaling (393.13s)
PASS
```

**Review by stephybun — APPROVED** [review]
Thanks @puneetsarna LGTM 👍
